pkg/corecli/extract.go

```diff
@@ -0,0 +1,62 @@
+package corecli
+
+import (
+	"github.com/dcos/dcos-cli/pkg/fsutil"
+	"github.com/dcos/dcos-cli/pkg/plugin"
+	"github.com/pelletier/go-toml"
+	"github.com/spf13/afero"
+	"path/filepath"
+)
+
+// TempPlugin will read the bundled corecli into a memmapfs and return a fake plugin from that data.
+func TempPlugin() (*plugin.Plugin, error) {
+	tempFs := afero.NewMemMapFs()
+	plugin := &plugin.Plugin{}
+
+	bundle, err := extractPlugin(tempFs)
+	if err != nil {
+		return nil, err
+	}
+
+	err = fsutil.Unzip(tempFs, bundle, "/dcos-core-cli")
+	if err != nil {
+		return nil, err
+	}
+
+	pluginFilePath := filepath.Join("/dcos-core-cli", "plugin.toml")
+	data, err := afero.ReadFile(tempFs, pluginFilePath)
+	if err != nil {
+		return nil, err
+	}
+	err = toml.Unmarshal(data, plugin)
+	if err != nil {
+		return nil, err
+	}
+
+	return plugin, nil
+}
+
+func extractPlugin(fs afero.Fs) (string, error) {
+	pluginData, err := Asset("core.zip")
+	if err != nil {
+		return "", err
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	// Write out the data into a temp directory so that it's in the real filesystem for buildPlugin
```

Comment:
Is it in the real filesystem or in the mem map fs? More generally does anything gets written to the real filesystem with these changes?

Reply:
I copied this from the code in `setup` and forgot to remove this comment. It doesn't reach into the real filesystem (though this function would if passed an Fs that does).